### DISCUSSION TITLE
Wild card subdomain to subfolder/path mapping

### DISCUSSION THREAD

**Discussion by smblee** [vercel/vercel#5700] [General] [answered]
Hello - I just discovered that Vercel had a wildcard domain feature which is super cool.

I have a use case with this wildcard and I'll use reddit as an example app to explain. 

Say I have `www.reddit.com` as my main app but also I have `*.reddit.com` which could be given to subreddits (e.g. `askreddit.reddit.com`, `buildapcsales.reddit.com`, etc...)

What I would like is to support 2 different behaviors for my main app (at www.) but also for my subdomain apps...

For all subdomains (except www),
`askreddit.reddit.com/posts/123` should map to the content that would be shown `www.reddit.com/r/askreddit/posts/123`

But obviously for apex subdomain (`www`), 

`www.reddit.com/posts/123` should map to a Not Found page since `/posts/[postId]` doesn't exist - only `/r/[subreddit]/posts/[postId]` does. So in order to access the same page, you would have to visit `www.reddit.com/r/askreddit/posts/123`

All the docs I have seen seem to only work either by using a custom server or a reverse proxy (which then I can't leverage Vercel anymore...), Or use vercel's redirect/rewrite feature (https://vercel.com/knowledge/can-i-redirect-from-a-subdomain-to-a-subpath) but this doesn't support wildcards base paths and I would have to explicitly state every "subreddit" I have in this mapping file.

Any insight on this would be appreciated.

**Comment by taylorhayduk** [comment]
I'd love to do this too.  For example, in getStaticPaths it would be nice to also have a "subdomain" property as well as the paths properties. 

My use case is powering multiple subdomains with CMS content.

**Comment by smblee** [comment]
Bumping this thread once!

**Comment by smblee** [comment, marked as answer]
Looks like this is supported now: https://github.com/vercel/vercel/discussions/5489#discussioncomment-1609713
